### PR TITLE
Support new PR view in `isMergedPR` and `isClosedPR`

### PR DESCRIPTION
**Pull request by kovsu**

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -288,8 +288,24 @@ TEST: addTests('isQuickPR', [
...
 export const isDraftPR = (): boolean => exists('#partial-discussion-header .octicon-git-pull-request-draft');
 export const isOpenPR = (): boolean => exists('#partial-discussion-header :is(.octicon-git-pull-request, .octicon-git-pull-request-draft)');
-export const isMergedPR = (): boolean => exists('#partial-discussion-header .octicon-git-merge');
-export const isClosedPR = (): boolean => exists('#partial-discussion-header :is(.octicon-git-pull-request-closed, .octicon-git-merge)');
+export const isMergedPR = (): boolean => {
+	if (exists('#partial-discussion-header :is(.octicon-git-merge)')) {
+		return true;
+	}
+
+	const status = $('[class^="StateLabel"]')!.textContent;
```
Comment (fregante) [inline index.ts:296]: Can this be used in both versions instead of keeping the exists() call?

**Comment by kovsu** [comment]
Should I  update `isOpenPR` and `isDraftPR` too?

File: index.ts [hunk 2]
```diff
@@ -286,10 +286,22 @@ TEST: addTests('isQuickPR', [
 	'https://github.com/sindresorhus/refined-github/compare/test-branch?quick_pull=1',
 ]);
 
-export const isDraftPR = (): boolean => exists('#partial-discussion-header .octicon-git-pull-request-draft');
-export const isOpenPR = (): boolean => exists('#partial-discussion-header :is(.octicon-git-pull-request, .octicon-git-pull-request-draft)');
-export const isMergedPR = (): boolean => exists('#partial-discussion-header .octicon-git-merge');
-export const isClosedPR = (): boolean => exists('#partial-discussion-header :is(.octicon-git-pull-request-closed, .octicon-git-merge)');
+export const isDraftPR = (): boolean => ($('.State') ?? $('[class^="StateLabel"]'))!.textContent!.trim() === 'Draft';
+export const isOpenPR = (): boolean => {
+	const status = ($('.State') ?? $('[class^="StateLabel"]'))!.textContent!.trim();
```
Comment (fregante) [inline index.ts:291]: Why not join the selector in a single call?

Comment (kovsu) [reply index.ts:291]: 😮 New knowledge come into my head.

File: index.ts [hunk 3]
```diff
@@ -286,10 +286,27 @@ TEST: addTests('isQuickPR', [
...
 ]);
 
-export const isDraftPR = (): boolean => exists('#partial-discussion-header .octicon-git-pull-request-draft');
-export const isOpenPR = (): boolean => exists('#partial-discussion-header :is(.octicon-git-pull-request, .octicon-git-pull-request-draft)');
-export const isMergedPR = (): boolean => exists('#partial-discussion-header .octicon-git-merge');
-export const isClosedPR = (): boolean => exists('#partial-discussion-header :is(.octicon-git-pull-request-closed, .octicon-git-merge)');
+const prStateSelector = [
+	'.State',
+	'[class^="StateLabel"]',
+].join(',');
```
Comment (fregante) [inline index.ts:292]:
```suggestion
];
```

If it doesn't work, just update `select-dom`

**Review by fregante — APPROVED** [review]

File: index.ts (hunk 3 above)
Comment (kovsu) [reply index.ts:292]: image

image


image

Comment (fregante) [reply index.ts:292]: I'm referring to dropping `.join()`

Comment (kovsu) [reply index.ts:292]: But it doesn't support string[].

image

Comment (fregante) [reply index.ts:292]: Ah yeah because we're not using `select-dom`

**Comment by fregante** [comment]
Can you also add support for isClosedIssue and any related ones?

**Comment by kovsu** [comment]
@fregante 👌

**Comment by kovsu** [comment]
image

**Pull request merged**